pkg/kritis/crd/securitypolicy/securitypolicy.go

```diff
@@ -194,3 +239,101 @@ func severityWithinThreshold(maxSeverity string, severity string) (bool, error)
 	}
 	return vulnerability.Severity_value[severity] <= vulnerability.Severity_value[maxSeverity], nil
 }
+
+type Attestor struct {
+	Name       string
+	PublicKeys []*AttestorPublicKey
+}
+
+type AttestorPublicKey struct {
+	ID          string
+	AsciiArmor  string
+	Fingerprint string
+}
+
+type AttestorFetcher interface {
+	GetAttestor(name string) (*Attestor, error)
+}
+
+type binauthzAttestorFetcher struct {
+	client binauthz.Client
+}
+
+func NewAttestorFetcher() (AttestorFetcher, error) {
+	client, err := binauthz.New()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create a binauthz client")
+	}
+	return &binauthzAttestorFetcher{
+		client: client,
+	}, nil
+}
+
+func (f *binauthzAttestorFetcher) GetAttestor(name string) (*Attestor, error) {
+	a, err := f.client.GetAttestor(context.Background(), name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get an attestor: %s", name)
+	}
+	if a.UserOwnedDrydockNote == nil {
+		return nil, errors.Wrapf(err, "attestor doesn't have UserOwnedDrydockNote: %s", name)
+	}
+
+	pubKeys := []*AttestorPublicKey{}
+	for _, pubKey := range a.UserOwnedDrydockNote.PublicKeys {
+		fingerprint, err := fingerprint(pubKey.AsciiArmoredPgpPublicKey)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get fingerprint of PGP public key: %s", pubKey.Id)
+		}
+		pubKeys = append(pubKeys, &AttestorPublicKey{
+			ID:          pubKey.Id,
+			AsciiArmor:  pubKey.AsciiArmoredPgpPublicKey,
+			Fingerprint: fingerprint,
+		})
+	}
+
+	attestor := &Attestor{
+		Name:       name,
+		PublicKeys: pubKeys,
+	}
+
+	return attestor, nil
+}
+
+func fingerprint(key string) (string, error) {
+	s, err := secrets.NewPgpKey("", "", key)
+	if err != nil {
+		return "", err
+	}
+	return s.Fingerprint(), nil
+}
+
+func NewBinauthzAttestorFetcher() (AttestorFetcher, error) {
```

Comment:
This function is not used anywhere and duplicated with `NewAttestorFetcher`.

Reply:
Oops, good catch.
Fixed in d586751